client/ayon_maya/plugins/publish/collect_ornatrix_rig.py

```diff
@@ -0,0 +1,101 @@
+import os
+from typing import List, Dict, Any
+import pyblish.api
+from ayon_core.pipeline.publish import KnownPublishError
+from ayon_maya.api import lib
+from ayon_maya.api import plugin
+from maya import cmds
+
+
+ORNATRIX_NODES = {
+    "HairFromGuidesNode", "GuidesFromMeshNode",
+    "MeshFromStrandsNode", "SurfaceCombNode"
+}
+
+
+class CollectOxRig(plugin.MayaInstancePlugin):
+    """Collect all information of the Ornatrix Rig"""
+
+    order = pyblish.api.CollectorOrder + 0.4
+    label = "Collect Ornatrix Rig"
+    families = ["oxrig"]
+
+    def process(self, instance):
+        ornatrix_nodes = cmds.ls(instance.data["setMembers"], long=True)
+        self.log.debug(f"Getting ornatrix nodes: {ornatrix_nodes}")
+
+        ornatrix_resources = []
+        ornatrix_nodes_list = []
+
+        # Use `set` to avoid duplicate resource data
+        for node in set(ornatrix_nodes):
+            # Get Yeti resources (textures)
+            resources = self.get_texture_resources(node)
+            ornatrix_resources.extend(resources)
+
+        instance.data["resources"] = ornatrix_resources
+        self.log.debug(instance.data["resources"])
+        for node in ornatrix_nodes:
+            ox_node_list = self.get_ox_nodes(node)
+            ornatrix_nodes_list.extend(ox_node_list)
+
+        instance.data["ornatrix_nodes"] = ornatrix_nodes_list
+        self.log.debug(instance.data["ornatrix_nodes"])
+
+    def get_ox_nodes(self, node: str) -> List[Dict[str, Any]]:
+        all_ox_nodes = []
+        node_shape = cmds.listRelatives(node, shapes=True)
+        if not node_shape:
+            return []
+
+        ox_nodes = cmds.ls(
+            cmds.listConnections(node_shape, destination=True) or [],
+            type=ORNATRIX_NODES)
+        if ox_nodes:
+            all_ox_nodes.append(node)
+        return all_ox_nodes
```

Comment:
This logic seems wrong @moonyuet 

It's returning the the original `[node]` if it happens to be connected to any ornatrix nodes. It's not returning the ornatrix nodes? Either function name is wrong or confusing.. or we have a bug?

Currently on Line 38-40 we're now just extending the list we're iterating putting in the same nodes, again and again. Which may even end up generating an infinite recursion?

Reply:
Resolved with https://github.com/ynput/ayon-maya/pull/1/commits/0984b91f5c8188ed9b636f85cdb8175ef4067f8e